rdt/transformers/pii/anonymizer.py

```diff
@@ -209,3 +214,148 @@ def __repr__(self):
+
+    This transformer anonymizes values that can be traced back to the original input by using
+    a mapping. The transformer will generate a mapping with the previously specified
+    ``Faker`` provider and ``function``.
+
+    Args:
+        provider_name (str):
+            The name of the provider in ``Faker``. If ``None`` the ``BaseProvider`` is used.
+            Defaults to ``None``.
+        function_name (str):
+            The name of the function to use within the ``faker.provider``. Defaults to
+            ``lexify``.
+        function_kwargs (dict):
+            Keyword args to pass into the ``function_name`` when being called.
+        locales (list):
+            List of localized providers to use instead of the global provider.
+    """
+
+    def __getstate__(self):
+        """Return a dictionary representation of the instance and warn the user when pickling."""
+        LOGGER.warning((
+            'You are saving the mapping information, which includes the original data. '
+            'Sharing this object with others will also give them access to the original data '
+            'used with this transformer.'
+        ))
+
+        return self.__dict__
+
+    def __init__(self, provider_name=None, function_name=None, function_kwargs=None, locales=None):
+        super().__init__(
+            provider_name=provider_name,
+            function_name=function_name,
+            function_kwargs=function_kwargs,
+            locales=locales,
+        )
+
+        self._mapping_dict = {}
+        self._reverse_mapping_dict = {}
+        self._label_encoder = None
+
+    def get_mapping(self):
+        """Return the mapping dictionary."""
+        return deepcopy(self._mapping_dict)
+
+    def get_output_sdtypes(self):
+        """Return the output sdtypes supported by the transformer.
+
+        Returns:
+            dict:
+                Mapping from the transformed column names to supported sdtypes.
+        """
+        output_sdtypes = {'value': 'float'}
+        return self._add_prefix(output_sdtypes)
+
+    def _fit(self, columns_data):
+        """Fit the transformer to the data.
+
+        Generate a ``_mapping_dict`` and a ``_reverse_mapping_dict`` for each
+        value in the provided ``columns_data`` using the ``Faker`` provider and
+        ``function``.
+
+        Args:
+            data (pandas.Series):
+                Data to fit the transformer to.
+        """
+        super()._fit(columns_data)
+        unique_values = columns_data[columns_data.notna()].unique()
+        unique_data_length = len(unique_values)
+        generated_values = [self._function() for _ in range(unique_data_length)]
+        if len(set(generated_values)) != unique_data_length:
+            counter = 0
+            while len(set(generated_values)) != unique_data_length:
+                generated_values.append(self._function())
+                counter += 1
+                if counter == 10:
```

Comment:
Why is this capped at 10?

Reply:
yeah, if this is an arbitrary number can we set it as a class attribute like `MAX_ATTEMPTS_FOR_UNIQUE_VALUES` or something like that

Reply:
Yes this was a little bit 'improvised'. This piece of the code could change in a different ways, and I also had different ideas.

The current one is basically adding new values to `generated_values` and tries 10 times, if it fails that amount of times (which I agree we can set a `MAX_ATTEMPTS_FOR_UNIQUE_VALUES`), the other approach is to generate the same amount of data and see if the `set` is `>= unique_data_length` and then just use `list(set(generated_values))[:unique_data_length]` . (I feel that the second one is less efficient when working with huge amounts of data that's why I decided to go with the first option).